### PR TITLE
All players now have a 2% chance of spawning with a random disability (with a small chance of power on top)

### PR DESCRIPTION
**Pull request by DeityLink**
Alternative to #29869

If one PR gets merged I'll close the other one. Don't merge both (or they'll conflict anyway)

Originally suggested by @DamianX 

:cl:
* experiment: Instead of players having a 20% chance of spawning with "dormant" disabilities that would activate when any other gene would activate (or in the case of Bartenders and Gumshoes, immediately upon spawn), all players now have 2% chance of spawning with a genetic disability, in which case players have an additional 10% chance of getting a beneficial gene on top.

### CODE REVIEW

**Comment by Jackreturnswithaveangenace** [comment]
I'm very torn on this. I like both PRs as an alternative to the current situation, but even then the idea of dormant genes being used intentionally at some point seems interesting too.

**Comment by SonixApache** [comment]
i stand with italy
looking forward to the server vote

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -2026,6 +2026,19 @@ mob/living/carbon/human/isincrit()
...
+		dna.GiveRandomSE(notflags = GENE_UNNATURAL,genetype = GENETYPE_BAD)
+		if(prob(chanceForGoodIfBad))
+			dna.GiveRandomSE(notflags = GENE_UNNATURAL,genetype = GENETYPE_GOOD)
+
+	if(prob(goodGeneProb))
+		dna.GiveRandomSE(notflags = GENE_UNNATURAL,genetype = GENETYPE_GOOD)
+		if(prob(chanceForBadIfGood))
+			dna.GiveRandomSE(notflags = GENE_UNNATURAL,genetype = GENETYPE_BAD)
+
+	check_mutations = 1
```
Comment (d3athrow) [inline code/modules/mob/living/carbon/human/human.dm:2040]: I suggest you leave the mutations dormant

**Comment by DeityLink** [comment]
image

My thoughts as copied from the previous PR:

> You see, players were intended to have 20% chance to spawn (roundstart or latejoin) with a genetic disability. And you thought 5% roundstart virus chance was bad. However due to the way gene code works, domutcheck() has to be called for genes to actually activate after SE are set. This not being done, players would carry "dormant" disabilities that would then activate as soon as something else triggered domutcheck, such as a dna injector, a virus that triggers a genetic power/disability, even lings using transform. But the most interesting part is that Gumshoes and Bartenders, who DO spawn with extra active genes end up being the only jobs that have the 20% chance of actually having a disability on top. If you ever latejoined blind, or clumsy, etc, as a bartender, this is why. And this was originally intended to be a possibility for ALL players. **This sucks**. Let's save bartenders and gumshoes from this hell, and **fix bugs related to this unintended behaviour**.

Some of those bugs having been reported in the past by tickets such as #28469 or #6920

From there, there are 4 solutions:

1. We ignore the issue altogether, and close the above bugs as "working as intended"
2. We try and band-aid the feature so it works like before except it doesn't activate at roundstart for Bartenders and Gumshoes automatically (and possibly lower the chance, Pomf's suggestion).
3. We have the feature affect all players equally by no longer allowing genes to stay dormants, but have it work at a lower chance (**THIS PR**)
4. We remove the possibility of roundstart disabilities, dormant or not (**THE PREVIOUS PR**)

I lean heavily toward 4., but I'd like to do a server vote to gauge the playerbase's general opinion on the matter.

**Comment by d3athrow** [comment]
move da fuck up genes to after the player has received their equipment because jobbies mutations are set in outfit/./post_equip for some reason
https://github.com/vgstation-coders/vgstation13/blob/bebba36718bc843f1aaa4071c9cc339410203ac1/code/game/gamemodes/gameticker.dm#L426 btw
edit: also need to make fuckupgenes not a child of new_player

**Comment by d3athrow** [comment]
2nd comment anyways to keep the thoughts separate. 

The feature works completely as intended for late spawn players as fuckupgenes is called after they have been equipped per their job.

**Comment by Eneocho** [comment]
Good idea so long as it isn't a roundstart mute/blind chance.

**Comment by BomberBro** [comment]
Not much to say on this PR besides the idea being dumb as all hell. 2% chance to have blindness, deafness, mute or seizures roundstart does not make for a good feature.

**Comment by DeityLink** [comment]
> I lean heavily toward 3

Shit I meant 4. Also I prefer Bathos' PR personally. Any solution that fixes the issue is good to me but

**Comment by DeityLink** [comment]
Yeah no I hate it. Even with a 2% chance I don't want to force that on the crew.

Someone else copy this PR and open it yourself if you really want random roundstart disabilities.

**Pull request closed without merging**